### PR TITLE
add HasCallStack to a few yesod test utilities that call throwString

### PR DESCRIPTION
**Pull request by chris-martin**

### CODE REVIEW

File: freckle-app/library/Freckle/App/Test/Yesod.hs [hunk 1]
```diff
@@ -159,20 +159,24 @@ testSetCookie :: forall m site. MonadYesodExample site m => SetCookie -> m ()
 testSetCookie = liftYesodExample . Yesod.Test.testSetCookie
 
 -- | Get the body of the most recent response and decode it as JSON
-getJsonBody :: forall a m site. (MonadYesodExample site m, FromJSON a) => m a
+getJsonBody
+  :: forall a m site. (MonadYesodExample site m, FromJSON a, HasCallStack) => m a
 getJsonBody = either err pure . eitherDecode =<< getRawBody
  where
   err e = throwString $ "Error decoding JSON response body: " <> e
```
Comment (pbrisbin) [inline freckle-app/library/Freckle/App/Test/Yesod.hs:166]: Drive-by, but I think it'd be nice if these used `expectationFailure` instead of `throwString`.

**Review by pbrisbin — APPROVED** [review]

File: freckle-app/library/Freckle/App/Test/Yesod.hs (hunk 1 above)
Comment (chris-martin) [reply freckle-app/library/Freckle/App/Test/Yesod.hs:166]: Thank you, I think this might actually just be the solution to the problem I have (not seeing useful output when a test fails from this throw).
